service/src/main/java/bio/terra/workspace/app/controller/WorkspaceV2Api.java

```diff
@@ -0,0 +1,222 @@
+import bio.terra.workspace.generated.model.ApiCloudPlatform;
+import bio.terra.workspace.generated.model.ApiCreateWorkspaceV2Request;
+import bio.terra.workspace.generated.model.ApiCreateWorkspaceV2Result;
+import bio.terra.workspace.generated.model.ApiDeleteCloudContextV2Request;
+import bio.terra.workspace.generated.model.ApiDeleteWorkspaceV2Request;
+import bio.terra.workspace.generated.model.ApiJobReport;
+import bio.terra.workspace.generated.model.ApiJobResult;
+import bio.terra.workspace.generated.model.ApiWorkspaceDescription;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequestFactory;
+import bio.terra.workspace.service.iam.SamService;
+import bio.terra.workspace.service.iam.model.SamConstants;
+import bio.terra.workspace.service.iam.model.WsmIamRole;
+import bio.terra.workspace.service.job.JobService;
+import bio.terra.workspace.service.spendprofile.SpendProfile;
+import bio.terra.workspace.service.spendprofile.SpendProfileId;
+import bio.terra.workspace.service.workspace.WorkspaceService;
+import bio.terra.workspace.service.workspace.exceptions.MissingRequiredFieldsException;
+import bio.terra.workspace.service.workspace.model.CloudPlatform;
+import bio.terra.workspace.service.workspace.model.Workspace;
+import bio.terra.workspace.service.workspace.model.WorkspaceStage;
+import java.util.Optional;
+import java.util.UUID;
+import javax.servlet.http.HttpServletRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+
+/**
+ * This class contains the implementation of the WorkspaceV2 endpoints. They are directly forwarded
+ * from the controller to here.
+ */
+@Component
+public class WorkspaceV2Api {
+  private static final Logger logger = LoggerFactory.getLogger(WorkspaceV2Api.class);
+  private final AuthenticatedUserRequestFactory authenticatedUserRequestFactory;
+  private final JobApiUtils jobApiUtils;
+  private final JobService jobService;
+  private final HttpServletRequest request;
+  private final SamService samService;
+  private final WorkspaceApiUtils workspaceApiUtils;
+  private final WorkspaceService workspaceService;
+
+  public WorkspaceV2Api(
+      AuthenticatedUserRequestFactory authenticatedUserRequestFactory,
+      JobApiUtils jobApiUtils,
+      JobService jobService,
+      HttpServletRequest request,
+      SamService samService,
+      WorkspaceApiUtils workspaceApiUtils,
+      WorkspaceService workspaceService) {
+    this.authenticatedUserRequestFactory = authenticatedUserRequestFactory;
+    this.jobApiUtils = jobApiUtils;
+    this.jobService = jobService;
+    this.request = request;
+    this.samService = samService;
+    this.workspaceApiUtils = workspaceApiUtils;
+    this.workspaceService = workspaceService;
+  }
+
+  public ResponseEntity<ApiCreateWorkspaceV2Result> createWorkspaceV2(
```

Comment:
@Traced and @override

Reply:
I added the `@Traced` in the WorkspaceApiController. The `@Override` was already there. I think that is the right spot for it.